tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/eval/ParseAnalyzer.scala

```diff
@@ -0,0 +1,116 @@
+package org.allenai.nlpstack.parse.poly.eval
+
+import org.allenai.nlpstack.parse.poly.polyparser.PolytreeParse
+
+/** A ParseAnalyzer maps a candidate parse to an "analysis", i.e. a histogram.
+  *
+  * For instance, this histogram may be the count of mistaken arclabels or part-of-speech tags.
+  */
+trait ParseAnalyzer extends (PolytreeParse => Map[String, Double]) {
+  def name: String
+}
+
+/** The MisattachmentAnalyzer tallies misattached tokens (i.e. tokens with the wrong breadcrumb
+  * assignment) according to the label of its breadcrumb arc in the gold parse.
+  *
+  * @param goldParseBank a bank containing the gold parses
+  * @param ignoreLabel set to true if we want to regard a node as correctly attached as long as
+  * its breadcrumb is correct (regardless of how its breadcrumb arc is labeled)
+  */
+case class MisattachmentAnalyzer(
+    goldParseBank: ParseBank,
+    ignoreLabel: Boolean
+) extends ParseAnalyzer {
+
+  def apply(candParse: PolytreeParse): Map[String, Double] = {
+    goldParseBank.askForGoldParse(candParse) match {
+      case Some(goldParse) =>
+        val validTokens = Range(1, goldParse.tokens.size) filter { tokIndex =>
+          goldParse.tokens(tokIndex).getDeterministicProperty('cpos) != Symbol(".")
```

Comment:
What's `cops == "."` ? A comment would be helpful. Are you just looking for a full stop? Since you're using question banks, sentences usually end in question marks though, so not sure what the above is...